Does not compile on the latest Nightly

On `rustc 1.27.0-nightly (ac3c2288f 2018-04-18)` compiling clippy yields:
```
   Compiling clippy_lints v0.0.194
error[E0308]: mismatched types
    --> /home/bemeurer/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.194/src/loops.rs:1801:48
     |
1801 |         ty::TyArray(_, n) => (0..=32).contains(n.val.to_raw_bits().expect("array length")),
     |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |                                                |
     |                                                expected reference, found u128
     |                                                help: consider borrowing here: `&n.val.to_raw_bits().expect("array length")`
     |
     = note: expected type `&_`
                found type `u128`

error[E0308]: mismatched types
   --> /home/bemeurer/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.194/src/regex.rs:213:56
    |
213 |                         str_span(expr.span, *e.span(), offset),
    |                                                        ^^^^^^ expected usize, found u16

error[E0308]: mismatched types
   --> /home/bemeurer/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.194/src/regex.rs:221:56
    |
221 |                         str_span(expr.span, *e.span(), offset),
    |                                                        ^^^^^^ expected usize, found u16

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0308`.

```

--- a/clippy_lints/src/loops.rs
+++ b/clippy_lints/src/loops.rs
@@ -1798,7 +1798,7 @@ fn is_ref_iterable_type(cx: &LateContext, e: &Expr) -> bool {
 fn is_iterable_array(ty: Ty) -> bool {
     // IntoIterator is currently only implemented for array sizes <= 32 in rustc
     match ty.sty {
-        ty::TyArray(_, n) => (0..=32).contains(n.val.to_raw_bits().expect("array length")),
+        ty::TyArray(_, n) => (0..=32).contains(&n.val.to_raw_bits().expect("array length")),
         _ => false,
     }
 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -12,7 +12,7 @@ extern crate clippy_lints;
 
 #[plugin_registrar]
 pub fn plugin_registrar(reg: &mut Registry) {
-    if let Ok(lint_store) = reg.sess.lint_store.try_borrow() {
+    reg.sess.lint_store.with_read_lock(|lint_store| {
         for (lint, _, _) in lint_store.get_lint_groups() {
             if lint == "clippy" {
                 reg.sess
@@ -21,7 +21,7 @@ pub fn plugin_registrar(reg: &mut Registry) {
                 return;
             }
         }
-    }
+    });
 
     clippy_lints::register_plugins(reg);
 }

